services/rfq/relayer/service/statushandler.go

```diff
@@ -224,6 +238,103 @@ func (q *QuoteRequestHandler) shouldCheckClaim(request reldb.QuoteRequest) bool
 	return true
 }
 
+// Cache based rate limiter.
+// We don't want to commit a very large balance of the relayer in one block straight away without confirmations.
+// Case 1: If the RFQ demands over $10k in one block, we should not relay the request unless we have a confirmation.
+// Case 2: We should also not relay the request if the cumulative relayed amount over the last blockWindowSize blocks
+// exceeds the volume limit AND we have not confirmed.
+func (q *QuoteRequestHandler) canRelayBasedOnVolumeAndConfirmations(
+	request reldb.QuoteRequest,
+	currentBlockNumber uint64,
+	volumeLimit float64,
+) (bool, error) {
+	// Case 1: Singular RFQ over volumeLimit and inadequate confirmations
+
+	priceOfOriginToken, err := q.getTokenPrice(context.Background(), request)
```

Comment:
Let's pass `ctx` in as an argument to `canRelayBasedOnVolumeAndConfirmations`. In general `context.Background()` should only really be used when the process is initially spun up